### PR TITLE
Draft, matchmaking, and database implementation

### PR DESCRIPTION
**Pull request by debashisbiswas**

### CODE REVIEW

File: pombot/lib/event_handlers.py [hunk 1]
```diff
@@ -26,3 +29,34 @@ async def on_message_handler(bot: Bot, message: Message):
...
+            viking_role = None
+            for role in guild.roles:
+                if role.name == Pomwars.KNIGHTS_ROLE_NAME:
+                    knight_role = role
+                if role.name == Pomwars.VIKINGS_ROLE_NAME:
+                    viking_role = role
+
+            if knight_role is not None and viking_role is not None:
+                await payload.member.remove_roles(knight_role, viking_role)
+                team = knight_role
```
Comment (clayboone) [inline pombot/lib/event_handlers.py:52]: Is this line needed? Maybe leftover from debug? It seems to be set in all future cases.

File: pombot/lib/event_handlers.py [hunk 2]
```diff
@@ -26,3 +29,34 @@ async def on_message_handler(bot: Bot, message: Message):
...
+
+                if payload.guild_id in Pomwars.KNIGHT_ONLY_GUILDS:
```
Comment (clayboone) [inline pombot/lib/event_handlers.py:54]: Moving these guild ID's around is going to be messy... I don't have a solution, just an observation.

File: pombot/lib/types.py [hunk 3]
```diff
@@ -93,3 +94,23 @@ class Event:
...
+
+
+class ActionType(str, Enum):
+    """Type of an action in the actions table of the database."""
+    NORMAL_ATTACK = 'normal_attack'
+    HEAVY_ATTACK = 'heavy_attack'
+    DEFEND = 'defend'
+
+
+class InstantItem(str, Enum):
```
Comment (clayboone) [inline pombot/lib/types.py:112]: Did we not rename this? Or is this the new name..

**Review by clayboone — CHANGES_REQUESTED** [review]
Finished review.

Consider almost all these comments optional. The only one I'm really concerned about is how the bot chooses which team to add a player to.

**Comment by debashisbiswas** [comment]
Updated to add the user to the database, with changes made based on most of the above comments.

**Pull request merged**